library/Test/Hspec/Core/Runner/Ext.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE CPP #-}
+
+-- | Compatibility module for 'configAvailableFormatters'
+--
+-- This feature is only available in hspec >= 2.9. Note that this module doesn't
+-- do anything to backport the feature; it just supplies a function that will
+-- use it when possible and safely no-op when not.
+--
+module Test.Hspec.Core.Runner.Ext
+  ( configAddAvailableFormatter
+  ) where
+
+import Prelude
+
+import Test.Hspec.Core.Format (Format, FormatConfig)
+import Test.Hspec.Core.Runner (Config(..))
+
+configAddAvailableFormatter
+  :: String -> (FormatConfig -> IO Format) -> Config -> Config
+#if MIN_VERSION_hspec_core(2,9,0)
```

Comment:
Ugh, CPP.

Reply:
I'm not sure we'll be happy with ourselves if we hold out for 2.9 and then support only it. But feel free to push for that.

The `--format`-style of enabling is a nice-to-have, so I could just drop it altogether.

Reply:
Yeah, I'm fine with the CPP for now. Though I'd like to kill it at some point.